Sources/PackageCollectionsSigning/Utilities/Base64URL.swift

```diff
@@ -0,0 +1,88 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2021 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+ */
+
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Vapor open source project
+//
+// Copyright (c) 2017-2020 Vapor project authors
+// Licensed under MIT
+//
+// See LICENSE for license information
+//
+// SPDX-License-Identifier: MIT
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+
+// Source: https://github.com/vapor/jwt-kit/blob/master/Sources/JWTKit/Utilities/Base64URL.swift
+
+extension DataProtocol {
+    func base64URLDecodedBytes() -> [UInt8] {
+        return Data(base64Encoded: Data(self.copyBytes()).base64URLUnescaped())?.copyBytes() ?? []
+    }
+
+    func base64URLEncodedBytes() -> [UInt8] {
+        return Data(self.copyBytes()).base64EncodedData().base64URLEscaped().copyBytes()
```

Comment:
A similar note applies here: this does 5 allocs and 4 copies too. Similar notes as above allow us to remove several of these allocations.

Reply:
Cleaned this up in https://github.com/apple/swift-package-manager/pull/3242/commits/667c95473df57c837a561043821750a42cf28691